### ISSUE TITLE
accuracy on test set decreased after loading the saved model in different terminal.

### ISSUE THREAD

**Issue by aniketbote** [keras-team/keras#12583]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

I have trained a classifier which gives 90% acc on testing set.
When I save and load and test it without closing the terminal I get 90% acc.
But when I close the terminal and restart it, load the saved model I get very bad accuracy.

My script:
```
from keras.models import Sequential
from keras.layers import Dense

model = Sequential()

model.add(Dense(units=64, activation='relu'))
model.add(Dense(units=32, activation='relu'))
model.add(Dense(units=2, activation='softmax'))

checkpoint_path = "/content/gdrive/My Drive/santander_weights/cp1-{epoch:04d}.ckpt"
checkpoint_dir = os.path.dirname(checkpoint_path)


# Create checkpoint callback
cp_callback = tf.keras.callbacks.ModelCheckpoint(
    checkpoint_path, verbose=1, save_weights_only=True,
    # Save weights, every 5-epochs.
    period=5)

model.compile(loss='sparse_categorical_crossentropy',
              optimizer='Adam',
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=50,callbacks = [cp_callback], batch_size=100)
model.save('1_3.h5')

classes = list(model.predict(x_test))
final = []
for i in classes:
  y_classes = i.argmax(axis=-1)
  final.append(y_classes)

from sklearn.metrics import f1_score
score = f1_score(y_test,final,average = None)
print(score)
```

score = [0.89871326 0.90193236]

When i load the model from another script:
script2:
```
import tensorflow as tf
lol = tf.keras.models.load_model('/content/gdrive/My Drive/1_3.h5')
classes = list(lol.predict(x_test))
final = []
for i in classes:
  #print(i)
  y_classes = i.argmax(axis=-1)
  #print(y_classes)
  final.append(y_classes)
print('done')
from sklearn.metrics import f1_score
score = f1_score(y_test,final,average = None)
print(score)
```
score = [0.50019757 0.40107532]

**Comment by benbuleong** [comment]
I have a similar issue, except that I am using a custom loss and metrics. I upgraded to tensorflow 1.13.1 as well as keras 2.2.4, but it is still weird: model.evaluate and model.predict using the same metrics and dataset should give the same results, but the difference is too great to be correct.
```
def f1(y_true, y_pred):
    y_pred = K.round(y_pred)
    tp = K.sum(K.cast(y_true*y_pred, 'float'), axis=0)
    tn = K.sum(K.cast((1-y_true)*(1-y_pred), 'float'), axis=0)
    fp = K.sum(K.cast((1-y_true)*y_pred, 'float'), axis=0)
    fn = K.sum(K.cast(y_true*(1-y_pred), 'float'), axis=0)

    p = tp / (tp + fp + K.epsilon())
    r = tp / (tp + fn + K.epsilon())

    f1 = 2*p*r / (p+r+K.epsilon())
    f1 = tf.where(tf.is_nan(f1), tf.zeros_like(f1), f1)
    return K.mean(f1)

def f1_loss(y_true, y_pred):
    
    tp = K.sum(K.cast(y_true*y_pred, 'float'), axis=0)
    tn = K.sum(K.cast((1-y_true)*(1-y_pred), 'float'), axis=0)
    fp = K.sum(K.cast((1-y_true)*y_pred, 'float'), axis=0)
    fn = K.sum(K.cast(y_true*(1-y_pred), 'float'), axis=0)

    p = tp / (tp + fp + K.epsilon())
    r = tp / (tp + fn + K.epsilon())

    f1 = 2*p*r / (p+r+K.epsilon())
    f1 = tf.where(tf.is_nan(f1), tf.zeros_like(f1), f1)
    return 1 - K.mean(f1)

callbacks = [EarlyStopping(monitor='val_f1', patience=15),
             ModelCheckpoint(filepath=dname+'_best_model.h5', monitor='val_f1', save_best_only=True, mode='max', verbose=1)]

opt = Adam(lr=0.0001, beta_1=0.9, beta_2=0.999, decay=0.01)
model.compile(loss=f1_loss, optimizer=opt, metrics=[f1])
model.fit([X_train_audio, X_train_video, X_train_text], Y_train, batch_size=128, epochs=30, callbacks=callbacks, validation_data=([X_test_audio, X_test_video, X_test_text],Y_test))

model.evaluate([X_test_audio, X_test_video, X_test_text],Y_test, batch_size=128, verbose=1)

Out: [0.6195135793534878, 0.5118402310086592]

predictions = model.predict([X_test_audio, X_test_video, X_test_text])
sess = tf.InteractiveSession()
print (sess.run(f1(Y_test,predictions)))

Out: 0.17375563
```

So, the `f1_loss` is 0.6195135793534878, while `f1` from evaluate is `0.5118402310086592` while that of predict is `0.17375563`. That's too large of a gap.

**Comment by sanheensethi** [comment]
Same Problem Occur with me , when i load my keras model , its accuracy is approx 3% and when i run this in same runtime its accuracy is about 88% 

My code is here : https://colab.research.google.com/drive/1jGJhzElkS4oWX1tLbT4KXHTYs3aoqQtj#scrollTo=nwyskCSfLXJc

if Someone have solution , please Reply.
